bokeh/embed.py

```diff
@@ -40,35 +40,73 @@ def components(plot_object, resources=None):
               are **already loaded**.
 
     Args:
-        plot_object (PlotObject) : Bokeh object to render
-            typically a Plot or PlotContext
+        plot_objects (PlotObject|list|dict|tuple) : Will return script
+        and div element to render if PlotObject, or correlating structures
+        of script/div combinations
         resources : Deprecated argument
     Returns:
-        (script, div) : UTF-8 encoded
-
+        (script, div[s]): UTF-8 encoded
     '''
-    
+
     if resources is not None:
         warn('Because the ``resources`` argument is no longer needed, '
              'is it deprecated and will be removed in'
              'a future version.', DeprecationWarning, stacklevel=2)
-    
-    ref = plot_object.ref
-    elementid = str(uuid.uuid4())
-    
+    all_models = []
+    plots = []
+    if isinstance(plot_objects, list) or isinstance(plot_objects, tuple):
+        divs = []
+        for idx, plot_object in enumerate(plot_objects):
+            elementid = str(uuid.uuid4())
+            _append_plot(all_models, plots, plot_object, elementid)
+            divs = _append_div(elementid, divs)
+        divs = tuple(divs) if isinstance(plot_objects, tuple) else divs
+        return _component_pair(all_models, plots, divs)
+    elif isinstance(plot_objects, dict):
+        divs = {}
+        for key in plot_objects.keys():
+            elementid = str(uuid.uuid4())
+            _append_plot(all_models, plots, plot_objects[key], elementid)
+            divs = _append_div(elementid, divs, key)
+        return _component_pair(all_models, plots, divs)
+    else:
+        # if single PlotObject
```

Comment:
in fact here... you are collecting all universe less list, tuple and dicts... and that universe is bigger than just the plot object... I know that the docstrings ask for plot_object or plot_object lists, tuples or dicts, but I can imagine people trying to pass numpy arrays... or maybe generators... and this will break... maybe a plot_object check is in place here... but maybe I am being too defensive...